app/assets/javascripts/drafts.coffee

```diff
@@ -158,20 +166,51 @@ $(document).ready ->
         labelFor = $(field).attr('for')
 
         if labelFor != undefined
-          labelFor = labelFor.slice(0, idIndex) + labelFor.slice(idIndex).replace(multipleIndex, multipleIndex + 1)
+          labelFor = labelFor.slice(0, idIndex) + labelFor.slice(idIndex).replace(multipleIndex, targetIndex)
           $(field).attr 'for', labelFor
 
       else if $(field).is('div')
         # also increment the id for data contacts divs
         id = $(field).attr('id')
-        id = id.slice(0, idIndex) + id.slice(idIndex).replace(multipleIndex, multipleIndex + 1)
+        id = id.slice(0, idIndex) + id.slice(idIndex).replace(multipleIndex, targetIndex)
         $(field).attr 'id', id
 
     newDiv
 
-  $('.metadata-form .multiple, .umm-form .multiple').on 'click', '.remove', ->
+  $('.metadata-form .multiple, .umm-form .multiple').on 'click', '.remove', (e) ->
     multipleItem = $(this).closest('.multiple-item')
-    $(multipleItem).remove()
+    if !multipleItem.siblings('.multiple-item').length
+      # Reset the form state to original page load.
+      # Adding another and deleting causes the formatting to break, so this is
+      # clearing out the existing fields instead.
+      # In order to return the document to the original state, the document needs
+      # to clear the Country/RelatedURLs before they would be cleared by changing
+      # the contact type, so go up the document in reverse
+      $(multipleItem.find('select:not([disabled="disabled"])').get().reverse()).each (index, element) ->
```

Comment:
so we needed to do the reverse after all?

Reply:
Different reverse.  There were two when I demoed it.  The second one was unnecessary, but this one remains so.